### PR TITLE
OCPBUGS-28658: bump k8s.io and others (master)

### PR DESCRIPTION
**Pull request by ibihim**
# What

Bump deps, in particular k8s.io

# Why

To pull http2 CVE fixes in.

Ref: https://github.com/openshift/oauth-apiserver/pull/91

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,96 +4,95 @@ go 1.20
...
-	github.com/openshift/api v0.0.0-20230620130318-b2e6ab3d6c69
-	github.com/openshift/apiserver-library-go v0.0.0-20230621110634-a99412818848
-	github.com/openshift/build-machinery-go v0.0.0-20230306181456-d321ffa04533
-	github.com/openshift/client-go v0.0.0-20230607134213-3cd0021bbee3
-	github.com/openshift/library-go v0.0.0-20230620084201-504ca4bd5a83
-	github.com/spf13/cobra v1.6.1
+	github.com/openshift/api v0.0.0-20231101013329-0d0d46454bb7
+	github.com/openshift/apiserver-library-go v0.0.0-20230503174907-d9b2bf6185e9
+	github.com/openshift/build-machinery-go v0.0.0-20220913142420-e25cf57ea46d
+	github.com/openshift/client-go v0.0.0-20230503144108-75015d2347cb
```
Comment (dgrisonnet) [inline go.mod:14]: these packages were downgraded for some reasons

File: pkg/cmd/oauth-apiserver/cmd.go [hunk 2]
```diff
@@ -17,6 +17,7 @@ import (
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	"k8s.io/apiserver/pkg/authorization/union"
 	"k8s.io/apiserver/pkg/endpoints/discovery/aggregated"
+	_ "k8s.io/apiserver/pkg/features"
```
Comment (dgrisonnet) [inline pkg/cmd/oauth-apiserver/cmd.go:20]: I don't think this is necessary. We should already be picking up the feature gates via other apiserver library imports.

File: go.mod (hunk 1 above)
Comment (ibihim) [reply go.mod:14]: Yeah, I took the dependencies of your PR, without checking if they are older. It is easier to miss it, if one doesn't use semver.

**Review by dgrisonnet — APPROVED** [review]
/lgtm

**Pull request merged**